spacy_readability/__init__.py

```diff
@@ -111,36 +116,55 @@ def smog(self, doc):
         """Returns the SMOG score for the document. If there are less than 30 sentences then
         it returns 0 because he formula significantly loses accuracy on small corpora.
         """
-        if self.num_sentences < 30 or self.num_words == 0:
+        num_sentences = self.get_num_sentences(doc)
+        num_words = self.get_num_words(doc)
+        if num_sentences < 30 or num_words == 0:
             return 0
         num_poly = self.get_num_syllables(doc, min_syllables=3)
-        return 1.0430 * sqrt(num_poly * 30 / self.num_sentences) + 3.1291
+        return 1.0430 * sqrt(num_poly * 30 / num_sentences) + 3.1291
 
     def coleman_liau(self, doc):
         """Returns the Coleman-Liau index for the document."""
-        letter_count = sum(
-            [len(token) for token in doc if not token.is_punct and not token.is_digit]
-        )
-        letters_to_words = letter_count / self.num_words * 100
-        sent_to_words = self.num_sentences / self.num_words * 100
-        return 0.0588 * letters_to_words - 0.296 * sent_to_words - 15.8
+        num_words = self.get_num_words(doc)
+        if num_words > 0:
+            num_sentences = self.get_num_sentences(doc)
+            letter_count = sum(
+                [len(token) for token in doc if not token.is_punct and not token.is_digit]
+            )
+            letters_to_words = letter_count / num_words * 100
+            sent_to_words = num_sentences / num_words * 100
+            return 0.0588 * letters_to_words - 0.296 * sent_to_words - 15.8
+        else:
+            return None
 
     def ari(self, doc):
         """Returns the Automated Readability Index for the document."""
-        letter_count = sum([len(token) for token in doc if not token.is_punct])
-        letter_to_words = letter_count / self.num_words
-        words_to_sents = self.num_words / self.num_sentences
-        return 4.71 * letter_to_words + 0.5 * words_to_sents - 21.43
+        num_sentences = self.get_num_sentences(doc)
+        num_words = self.get_num_words(doc)
+        if num_words > 0:
+            letter_count = sum([len(token) for token in doc if not token.is_punct])
+            letter_to_words = letter_count / num_words
+            words_to_sents = num_words / num_sentences
+            return 4.71 * letter_to_words + 0.5 * words_to_sents - 21.43
+        else:
+            return None
```

Comment:
```suggestion
             return 0
```
All the other methods return 0 when a pre-condition isn't met